packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/SimulationService.java

```diff
@@ -93,6 +93,8 @@ public Dataset createDatasetFromSimulation(
 		final String datasetName,
 		final UUID projectId,
 		final boolean addToProject,
+		final Optional<UUID> modelConfigurationId,
+		final Optional<UUID> interventionPolicyId,
```

Comment:
Similar to Derek's comment, you don't really need Optionals, just 
```
		final UUID modelConfigurationId,
		final UUID interventionPolicyId,
```
you can pass in nulls, Optionals are typically when you programmatically fetch something and you are not sure if you will have a result or not.